pkg/cli/verifier/envoy_config.go

```diff
@@ -470,6 +484,64 @@ func (v *EnvoyConfigVerifier) getDstMeshServicesForSvc(svc corev1.Service) ([]se
 	return meshServices, nil
 }
 
+func (v *EnvoyConfigVerifier) findIngressFilterChainForService(svc *corev1.Service, filterChains []*xds_listener.FilterChain) error {
+	if svc == nil {
+		return nil
+	}
+	src := v.configAttr.trafficAttr.SrcService
+	// grab endpoints for ingress service
+	endpoints, err := v.kubeClient.CoreV1().Endpoints(src.Namespace).Get(context.Background(), src.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("ingress source service %q not found: %w", src, err)
+	}
+
+	sourceIPs := map[string]bool{}
+	filterChainName := trafficpolicy.GetIngressTrafficMatchName(types.NamespacedName{
+		Name:      svc.GetName(),
+		Namespace: svc.GetNamespace(),
+	}, v.configAttr.trafficAttr.DstPort, v.configAttr.trafficAttr.AppProtocol)
+	var chain *xds_listener.FilterChain
+	for _, c := range filterChains {
+		if c.Name == filterChainName {
+			chain = c
+			break
+		}
+	}
+
+	if chain == nil {
+		return fmt.Errorf("ingress filter chain %s not found", filterChainName)
+	}
+
+	for _, ip := range chain.FilterChainMatch.SourcePrefixRanges {
+		if ip.PrefixLen.GetValue() == 32 {
+			sourceIPs[ip.AddressPrefix] = false
+		}
+	}
+
+	for _, sub := range endpoints.Subsets {
+		for _, ip := range sub.Addresses {
+			matched, ok := sourceIPs[ip.IP]
+			// Filter chain is missing a service endpoint
+			if !ok {
+				return fmt.Errorf("service endpoint %s was not found in the ingress inbound filter chain", ip.IP)
+			}
+			if ok && !matched {
+				// ingress endpoint found in filter chain match
+				sourceIPs[ip.IP] = true
+			}
+		}
+	}
+
+	for ip, matched := range sourceIPs {
+		// TODO: This situation may not actually be an error. Discuss
```

Comment:
That would be an error at that instant. If the source service (e.g. ingress controller) endpoint is not programmed on the ingress filter chain, it should be reported as an error.

Reply:
This is actually the inverse situation: the ingress filter chain has an IP that's NOT in the source service's endpoints

Reply:
Ah I see, yes that's not an error then, as it could correspond to a different source than the one specified in the command. Could you update the comment to reflect this?